File: CCC/2014/J2.java
import java.util.Scanner;

public class Main {
  public static void main(String[] args) {
    Scanner input = new Scanner(System.in);
    int v = input.nextInt();
    String s = input.next();
    int Acount = 0;
    int Bcount = 0;
    for(int i = 0; i < v; i++){
      String temp = s.substring(i,i+1);
      if (temp.equals("A")){
        Acount+=1;
      }else{
        Bcount+=1;
      } 
    }
    if (Acount < Bcount){
      System.out.println("B");
    }
    else if(Acount > Bcount){
      System.out.println("A");
    }
    else{
      System.out.println("Tie");
    }
    
    
  }
}

File: CCC/2008/J2.py
p = "ABCDE"

while True:


  b = int(input())

  n = int(input())

  if b == 4 and n == 1:

    break

  if b == 1:
    
    p = p[n%5:] + p[:n%5] 
  if b == 2:

    p = p[-n%5:] + p[:-n%5] 
    


    

  if b == 3:
    if n % 2 == 0:
      continue
    else:
      temp = p[0]
      p = p[1] + p[1:]
      p = p[0] + temp + p[2:]

ans = ""

for i in p:
  ans += i + " "

print(ans.strip())
